### ISSUE TITLE
Rollen in Jubla Organization Hierarchy (Rolle Extern)

### ISSUE THREAD

**Issue by richardjubla** [hitobito/hitobito_jubla#56]
Ich orientiere mich an den möglichen Rollen für die jubla.db gemäss Jubla Organization Hierarchy https://github.com/hitobito/hitobito_jubla#jubla-organization-hierarchy

Ich habe folgende Fragen zur Rolle "Extern"
```
Die Rolle Extern kann nur öffentliche Daten (Gruppen, Anlässe und Abos; keine anderen Personen) sehen.
Diese Rolle ist nur von Personen in dieser Ebene sichtbar, nicht von Personen aus darüber liegenden Ebenen.
```

**2x Extern in einfache Gruppe**
<img src="https://github.com/hitobito/hitobito_jubla/assets/95908537/91379cf3-faab-46a9-81a6-a98ab9a09cae" width="50%">
Die Rolle ist zwei Mal in der einfachen Gruppe aufgeführt. In der Applikation selbst kann sie nur ein mal ausgewählt werden. Das ist komisch, da ja dieser Teil "automatisch" erstellt wird und die realen Berechtigungen abbilden soll.

~~**Rolle nicht auswählbar/zuweisbar**
Obwohl die Rolle in jeder Ebene/Gruppe vertreten ist, kann sie in Gruppen teilweise nicht zugewiesen werden:
<img src="https://github.com/hitobito/hitobito_jubla/assets/95908537/578c9cd8-7489-40be-a588-8a6e600d2f80" width="50%">
<img src="https://github.com/hitobito/hitobito_jubla/assets/95908537/ca0db791-9916-4032-9497-107e6082f1ce" width="50%">
Beispiel: In jeder Schar (Gruppe) oder einer Fachgruppe kann die Rolle "Extern" nicht zugewiesen werden.~~

~~**Rolle nicht verwaltbar/sichtbar**
Falls es dennoch gelingt die Rolle zu "erlangen", zum Beispiel durch eine externe Anmeldung oder Elternzugang, ist diese Rolle für die zuständigen Gruppen/Ebenen nicht verwaltbar oder auch nicht sichtbar. 
<img src="https://github.com/hitobito/hitobito_jubla/assets/95908537/e89ddfae-306f-470f-9f1f-910a16ba98eb" width="50%">
Durch den Elternzugang (Externe Anmeldung) erhalten Eltern die Rolle "Extern", Kinder keine Rolle. 
Dies führt dazu, dass weder Eltern noch Kind durch die Gruppe/Ebene administriert werden kann. 
Mögliche Lösung: Durch die Verfügbarkeit der Rolle "Extern" sollten Eltern in jeder Ebene/Gruppe verwaltbar werden. Deren Kinder (ohne Rolle) werden über die Eltern sichtbar oder sollten ebenfalls die Rolle "Extern" oder "Kind" erhalten.~~
Siehe auch: https://github.com/hitobito/hitobito/issues/2125#issuecomment-1625347911 

Version: [Version 1.30.6](https://jubla.puzzle.ch/changelog)

**Comment by richardjubla** [comment]
Ich gehe davon aus, dass wir den Elternzugang https://github.com/hitobito/hitobito/issues/2196 nicht vollständig/optimal Testen können, sofern dieser Bug wie beschreiben besteht.

**Comment by RolandStuder** [comment]
@richardjubla Hast du sichergestellt, dass du auf der Ebene auf der du diese Zuweisungen auspropierst eine Rolle zugewiesen hat, die dir für diese Ebene Schreibrechte gewährt?

Du kannst diese Rolle Extern nur zuweisen, wenn du auf dieser Ebene selbst zugewiesen bist als eine Person mit Schreibrechten. Wenn du deine Schreibrechte aus einer übergeordneten Ebene hast, wirst du diese Rolle nicht zuweisen sehen können.

**Comment by richardjubla** [comment]
> @richardjubla Hast du sichergestellt, dass du auf der Ebene auf der du diese Zuweisungen auspropierst eine Rolle zugewiesen hat, die dir für diese Ebene Schreibrechte gewährt?

Nein, das habe ich nicht. Danke für diesen Hinweis. Ich kann bestätigen, dass in den von mir getesteten drei Fällen die Zuweisung der Rolle "Extern" funktioniert, sofern die dieses Profil die nötigen Berechtigungen dazu hat. Die Rolle ist demzufolge  auswählbar/zuweisbar. 

- Die Rolle ist zwei Mal in der einfachen Gruppe aufgeführt
![image](https://github.com/hitobito/hitobito_jubla/assets/95908537/d85540cb-22c1-4f79-aafd-a9ae3aaf91e6)
- Die Rolle ist wählbar/zuweisbar (sofern die Berechtigungen dazu stimmen) 
- Rolle ist verwaltbar/sichtbar: Aber eben korrekterweise nur für die berechtigten Profile

Zusammenhang zum Elternzugang: Falls durch eine externe Anmeldung die Rolle "Extern" für den Verwalter zugeteilt wird, ist dieses Profil durch die Schar (Rolle Leitung / Adressverwaltung) sichtbar. Obwohl sichtbar und die (Schreib)Rechte vorhanden sind, ist aber die Beziehung zwischen Verwalter - Kind nicht sichtbar. 
Für das berechtigte Profil ist kein Verwalter-Account sichtbar, das Kind (ohne Rolle) nicht verwertbar (Workaround: Kind in Datenbank suchen und zu eigener Schar/Ebene hinzufügen)


![image](https://github.com/hitobito/hitobito_jubla/assets/95908537/978c0e21-0128-48e4-8b7b-372e829c4592)
![image](https://github.com/hitobito/hitobito_jubla/assets/95908537/0840e00a-f114-4901-bb39-ac4f6500e23e)
![image](https://github.com/hitobito/hitobito_jubla/assets/95908537/c93ea174-5be2-4b09-84a8-b5ac87522e77)

**Comment by janjubla** [comment]
- Doppelte Aufführung der Rolle Extern unter Global > einfache Gruppe (https://github.com/hitobito/hitobito_jubla#jubla-organization-hierarchy) nicht reproduzierbar. Scheint mittlerweile geändert worden zu sein. (~22.09.2026)
- Zusammenhang mit Elternzugang nicht überprüft, Issue nicht verstanden. (~22.09.2026)
